source/opt/instruction.cpp

```diff
@@ -499,5 +499,51 @@ std::ostream& operator<<(std::ostream& str, const ir::Instruction& inst) {
   return str;
 }
 
+bool Instruction::HasSideEffects() const {
```

Comment:
There is a IsCombinatorInstruction() function which is effectively the reverse of this list. Our preference is to whitelist operations instead of blacklisting them so that future extensions are not buggy until fixed.

Reply:
You might want to look at the value numbering code to see the conditions used there.  There will need to be more than just looking for combinators.